File: demos/jans-chip/android/app/src/main/java/io/jans/chip/model/SSARegRequest.kt
package io.jans.chip.model

import com.google.gson.annotations.SerializedName

data class SSARegRequest (
    @SerializedName("client_name")
    var clientName: String? = null,

    @SerializedName("evidence")
    var evidence: String? = null,

    @SerializedName("jwks")
    var jwks: String? = null,

    @SerializedName("scope")
    var scope: String? = null,

    @SerializedName("response_types")
    var responseTypes: List<String?>,

    @SerializedName("grant_types")
    var grantTypes: List<String?>,

    @SerializedName("software_statement")
    var ssa: String? = null,

    @SerializedName("application_type")
    var applicationType: String? = null,

    @SerializedName("redirect_uris")
    var redirectUris: List<String?>,
)

File: demos/jans-chip/ios/Jans-Chip/Models/SSARegRequest.swift
//
//  SSARegRequest.swift
//  Jans-Chip
//
//  Created by Nazar Yavornytskyi on 26.07.2024.
//

import Foundation

public struct SSARegRequest: Codable, ErrorHandler {
    
    var errorMessage: String?
    var isSuccess: Bool = true
    
    private enum CodingKeys: String, CodingKey {
        case clientName = "client_name"
        case evidence
        case jwks
        case scope
        case responseTypes = "response_types"
        case grantTypes = "grant_types"
        case ssa = "software_statement"
        case applicationType = "application_type"
        case redirectUris = "redirect_uris"
    }
    
    var clientName: String?
    var evidence: String?
    var jwks: String?
    var scope: String?
    var responseTypes: [String]?
    var grantTypes: [String]?
    var ssa: String?
    var applicationType: String?
    var redirectUris: [String]?
}
